swift/ql/lib/codeql/swift/elements/decl/DeclWithMembers.qll

```diff
@@ -0,0 +1,26 @@
+private import codeql.swift.elements.decl.IterableDeclContext
+private import codeql.swift.elements.decl.VarDecl
+private import codeql.swift.elements.decl.SubscriptDecl
+private import codeql.swift.elements.decl.Decl
+
+class DeclWithMembers extends Decl, IterableDeclContext {
+  /**
+   * Gets the `index`th member (0-based) of this nominal type or extension `Decl`,
+   * including `AccessorDecl`s of immediate members.
+   */
+  override Decl getImmediateMember(int index) {
```

Comment:
Not sure, but maybe this logic should be part of the extractor or codegen.

Reply:
We generally prefer to put logic in QL when we have the choice, and there are no good reasons for putting in the extractor.

Reply:
... I'd like to tell you why we prefer that.  I think it has to do with QL being easier to see and easier to change.